cmd/tunneld/tunneld.go

```diff
@@ -95,7 +105,9 @@ func main() {
 			fatal("failed to start HTTPS: %s", s.ListenAndServeTLS(opts.tlsCrt, opts.tlsKey))
 		}()
 	}
-
+	if opts.letsEncrypt {
+		go startAutocert(opts, server, logger)
```

Comment:
Go routine here is not needed. 

Maybe some elements of this function could be taken up here.